test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -393,6 +433,30 @@ TEST(ParticipantTests, LookupTopicDescription)
     ASSERT_EQ(participant->lookup_topicdescription(topic_name), nullptr);
 }
 
+TEST(ParticipantTests, DeleteTopicInUse)
```

Comment:
We need to check also in this test the same with data writer, either in this pull request if #1155 is merged before or in the other if this one is merged before.